client-react/src/pages/app/functions/app-keys/HostKeys.tsx

```diff
@@ -196,20 +219,40 @@ const HostKeys: React.FC<HostKeysProps> = props => {
         return <></>;
       }
       return (
-        <TooltipHost
-          content={t('delete')}
-          id={`app-keys-host-keys-delete-tooltip-${index}`}
-          calloutProps={{ gapSpace: 0 }}
-          closeDelay={500}>
-          <IconButton
-            className={defaultCellStyle}
-            disabled={readOnlyPermission}
-            id={`app-settings-application-settings-delete-${index}`}
-            iconProps={{ iconName: 'Delete' }}
-            ariaLabel={t('delete')}
-            onClick={() => deleteHostKey(itemKey)}
-          />
-        </TooltipHost>
+        <div>
+          <TooltipHost
+            content={t('delete')}
+            id={`app-keys-host-keys-delete-tooltip-${index}`}
+            calloutProps={{ gapSpace: 0 }}
+            closeDelay={500}>
+            <IconButton
+              className={defaultCellStyle}
+              disabled={readOnlyPermission}
+              id={`app-settings-application-settings-delete-${index}`}
+              iconProps={{ iconName: 'Delete' }}
+              ariaLabel={t('delete')}
+              onClick={() => deleteHostKey()}
+            />
+          </TooltipHost>
+          <Callout
+            hidden={!isDeleteConfirmationDialogVisible}
+            onDismiss={() => onCalloutDismiss(itemKey)}
+            setInitialFocus={true}
+            target={`#${`app-settings-application-settings-delete-${index}`}`}>
+            <div className={hostKeyDeleteConfirmDialogInnerDivStyle}>
+              {t('functionHostKeyDeleteConfirmMessage')}
+              <div>
+                <PrimaryButton
+                  id={`confirmDeleteButton`}
+                  className={hostKeyDeleteConfirmButtonStyle}
+                  onClick={() => onCalloutDismiss(itemKey)}>
+                  {t('continue')}
+                </PrimaryButton>
+                <DefaultButton text={t('cancel')} onClick={() => onCancelDelete()} />
```

Comment:
Bind directly to `onCancelDelete`.

```suggestion
                <DefaultButton text={t('cancel')} onClick={onCancelDelete} />
```